src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/workspace/DynamicPersistentVolumeClaimWorkspaceVolume.java

```diff
@@ -0,0 +1,65 @@
+package org.csanchez.jenkins.plugins.kubernetes.volumes.workspace;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public class DynamicPersistentVolumeClaimWorkspaceVolume extends WorkspaceVolume {
+    private String storageClassName;
+    private String requestsSize;
+    private String accessModes;
+    @CheckForNull
+    private Boolean readOnly;
+
+    @DataBoundConstructor
+    public DynamicPersistentVolumeClaimWorkspaceVolume(String storageClassName, Boolean readOnly,
+                                                       String requestsSize, String accessModes) {
+        this.storageClassName = storageClassName;
+        this.requestsSize = requestsSize;
+        this.accessModes = accessModes;
+        this.readOnly = readOnly;
+    }
+
+    public String getAccessModes() {
+        return accessModes;
+    }
+
+    public String getRequestsSize() {
+        return requestsSize;
+    }
+
+    public String getStorageClassName() {
+        return storageClassName;
+    }
+
+    @Nonnull
+    public Boolean getReadOnly() {
+        return readOnly != null && readOnly;
+    }
+
+    @Override
+    public Volume buildVolume(String volumeName) {
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withNewPersistentVolumeClaim()
+                .withClaimName(null)
+                .withReadOnly(getReadOnly())
+                .and()
+                .build();
+    }
+
+    @Extension
+    @Symbol("dynamicPersistentVolumeClaimWorkspaceVolume")
```

Comment:
Check Javadoc: symbols should not include the supertype name. So at most `dynamicPersistentVolumeClaim`, or more typably `dynamicPVC`.